nf_core/utils.py

```diff
@@ -489,37 +489,44 @@ def get_biocontainer_tag(package, version):
         A ValueError, if the package name can not be found (404)
     """
 
-    def get_tag_date(tag_date):
-        # Reformat a date given by quay.io to  datetime
-        return datetime.datetime.strptime(tag_date.replace("-0000", "").strip(), "%a, %d %b %Y %H:%M:%S")
+    biocontainers_api_url = f"https://api.biocontainers.pro/ga4gh/trs/v2/tools/{package}/versions/{package}-{version}"
 
-    quay_api_url = f"https://quay.io/api/v1/repository/biocontainers/{package}/tag/"
+    def get_tag_date(tag_date):
+        """
+        Format a date given by the biocontainers API
+        Given format: '2021-03-25T08:53:00Z'
+        """
+        tag_date = tag_date.split("T")[0].split("-")
+        return datetime.date(year=int(tag_date[0]), month=int(tag_date[1]), day=int(tag_date[2]))
 
     try:
-        response = requests.get(quay_api_url)
+        response = requests.get(biocontainers_api_url)
     except requests.exceptions.ConnectionError:
-        raise LookupError("Could not connect to quay.io API")
+        raise LookupError("Could not connect to biocontainers.pro API")
     else:
         if response.status_code == 200:
-            # Get the container tag
-            tags = response.json()["tags"]
-            matching_tags = [t for t in tags if t["name"].startswith(version)]
-            # If version matches several images, get the most recent one, else return tag
-            if len(matching_tags) > 0:
-                tag = matching_tags[0]
-                tag_date = get_tag_date(tag["last_modified"])
-                for t in matching_tags:
-                    if get_tag_date(t["last_modified"]) > tag_date:
-                        tag = t
-                return package + ":" + tag["name"]
-            else:
-                return matching_tags[0]["name"]
+            print(response.json())
+            images = response.json()["images"]
+            singularity_image = None
+            docker_image = None
+            for img in images:
+                if img["image_type"] == "Docker":
+                    modification_date = get_tag_date(img["updated"])
+                    if docker_image and modification_date < get_tag_date(docker_image["updated"]):
+                        continue
+                    else:
+                        docker_image = img
+                if img["image_type"] == "Singularity":
+                    modification_date = get_tag_date(img["updated"])
+                    if singularity_image and modification_date < get_tag_date(singularity_image["updated"]):
+                        continue
+                    else:
+                        singularity_image = img
```

Comment:
Not sure that you need the `continue` statements here?

```suggestion
                    if docker_image and modification_date > get_tag_date(docker_image["updated"]):
                        docker_image = img
                elif img["image_type"] == "Singularity":
                    modification_date = get_tag_date(img["updated"])
                    if singularity_image and modification_date > get_tag_date(singularity_image["updated"]):
                        singularity_image = img
```

Reply:
I think this won't work because at first `docker_image` is `None` and so this will never get updated. But it might be possible to make this nicer, will think about it

Reply:
Changed that part of the code to make it more concise